lib/AST/Type.cpp

```diff
@@ -1221,6 +1221,7 @@ CanType TypeBase::getCanonicalType() {
 
     Result = GenericFunctionType::get(sig, inputTy, resultTy,
                                       function->getExtInfo());
+//    assert(function->hasError() || !Result->hasError());
```

Comment:
It looks like a nice assert but it made some compiler_crashers regress... so I'll look into this later.